src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -47,23 +48,37 @@ public class ServiceEndpoint
         /// </summary>
         public EndpointMetrics EndpointMetrics { get; internal set; } = new EndpointMetrics();
 
+        private ServiceEndpoint(EndpointType type, string name = "", string version = "1.0")
+        {
+            Name = name;
+            EndpointType = type;
+            Version = version;
+        }
+
         public ServiceEndpoint(string key, string connectionString) : this(connectionString)
         {
             (Name, EndpointType) = ParseKey(key);
         }
 
-        public ServiceEndpoint(string connectionString, EndpointType type = EndpointType.Primary, string name = "")
+        public ServiceEndpoint(TokenCredential credential,
```

Comment:
>ServiceEndpoint [](http://example.com/codeflow?start=15&length=15)

who needs this ctor? I'd prefer only having the required change

Reply:
@vicancy @terencefan please add this constructor back. It's extremely useful to allow working with Azure Identity providers directly instead of relying on connection string construction and parsing, which offers only a limited support for token credentials.